hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/assertions/ContractFnResultAsserts.java

```diff
@@ -67,8 +67,8 @@ public static Function<HapiSpec, Function<Object[], Optional<Throwable>>> isCont
             try {
                 assertEquals(1, actualObjs.length, "Extra contract function return values!");
                 String implicitContract = "contract" + rand.nextInt();
-                ContractID contract = TxnUtils.asContractId(Bytes.fromHexString(((Address) actualObjs[0]).toString())
-                        .toArray());
+                ContractID contract = TxnUtils.asContractId(
+                        ((Address) actualObjs[0]).toString().substring(2), spec);
```

Comment:
Wait, what's this `.substring(2)` all about?  Can it be trying to strip off a `0x` prefix??  If so, can't that just be encapsulated inside of `TxnUtils.asContractId`?

This - problematic - expression shows up in various places - as seen in this image (which also highlights how it _should_ be done if it was done only in the right places ...

(If it _isn't_ trying to strip off a `0x` that what is it doing?)

![image](https://github.com/hashgraph/hedera-services/assets/117694041/9f9ce452-161a-4e43-a9d5-7b1cfe44d918)